File: src/Helpers/ResponseBuilder.php
<?php

namespace App\Helpers;

class ResponseBuilder
{
    /**
     * Build success response
     */
    public static function success($message = 'Success', $data = [])
    {
        return [
            'success' => true,
            'message' => $message,
            'data' => $data
        ];
    }

    /**
     * Build error response
     */
    public static function error($message = 'Error', $code = 400)
    {
        http_response_code($code);
        return [
            'success' => false,
            'message' => $message,
            'code' => $code
        ];
    }

    /**
     * Return JSON response and exit
     */
    public static function json($response)
    {
        header('Content-Type: application/json');
        echo json_encode($response);
        exit;
    }
}

File: helpers/response_builder.py
from flask import jsonify

class ResponseBuilder:
    """Helper class for building standard API responses"""

    @staticmethod
    def success(message='Success', data=None):
        """Build success response"""
        if data is None:
            data = {}
        return {
            'success': True,
            'message': message,
            'data': data
        }

    @staticmethod
    def error(message='Error', code=400):
        """Build error response"""
        return {
            'success': False,
            'message': message,
            'code': code
        }, code

    @staticmethod
    def json_response(response, status_code=200):
        """Return JSON response"""
        if isinstance(response, tuple):
            return jsonify(response[0]), response[1]
        return jsonify(response), status_code
